src/js/components/PipelinesSidebar.jsx

```diff
@@ -1,13 +1,21 @@
 'use strict';
 var React = require('react'),
+    connect = require('react-redux').connect,
+    getIn = require('../util/immutable-utils').getIn,
     PipelineList = require('./pipelines/PipelineList'),
     assets = require('../util/assets'),
     Icon = require('./Icon'),
     model = require('../model');
 
+function mapStateToProps(reduxState) {
+  var pipelines = getIn(reduxState, 'pipelines.pipelines');
+  return {
+    pipelines: pipelines.toJS(),
```

Comment:
See kadam's comment in [this PR](https://github.com/vega/lyra/pull/345) about using `toJS` in `mapStateToProps`. Essentially, `toJS` can be an expensive function, so we want to use it as infrequently as possible. `mapStateToProps` should return something quick that can be reliably used in a straight equality test to determine whether to re-render the property.